src/utils/MainApi.js

```diff
@@ -0,0 +1,129 @@
+const BASE_URL = 'https://www.api.alexdar.news.students.nomoredomainssbs.ru';
+
+const processResponse = (res) => {
+  return res.ok
+    ? res.json()
+    : Promise.reject(`${res.status} ${res.statusText}`);
+};
+
+class MainApi {
+  register = ({ email, password, name }) => {
+    return fetch(`${BASE_URL}/signup`, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        password: password,
+        email: email,
+        name: name,
+      }),
+    }).then((res) => res.json());
```

Comment:
`      .then((res) => res.json())
`

You need to ckeck every response if it's `ok`



```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

This code is repeated in every request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can delete redundant repeating by only this line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the `reference` for the method rather than the call.